dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -101,117 +118,122 @@ def refute_estimate(self):
         identified_estimand.outcome_variable = ["dummy_outcome"]
 
-        sample_estimates = np.zeros(self._num_simulations)
         self.logger.info("Refutation over {} simulated datasets".format(self._num_simulations) )
         self.logger.info("The transformation passed: {}".format(self._transformations) )
 
-        groups = self.preprocess_data_by_treatment()
-        estimates = []
-        for key_train, _ in groups:
-            X_train = groups.get_group(key_train)[self._chosen_variables].values
-            new_outcome_train = groups.get_group(key_train)['y'].values
-            validation_df = [] 
-            for key_validation, _ in groups:
-                if key_validation != key_train:
-                    validation_df.append(groups.get_group(key_validation))
-
-            validation_df = pd.concat(validation_df)
-            X_validation = validation_df[self._chosen_variables].values
-            new_outcome_validation = validation_df['y'].values
-
-            for action, func_args in self._transformations:
-
-                if callable(action):
-                    estimator = action(X_train, new_outcome_train, **func_args)
-                    new_outcome_train = estimator(X_train)
-                    new_outcome_validation = estimator(X_validation)
-                elif action in DummyOutcomeRefuter.SUPPORTED_ESTIMATORS:
-                    estimator = self._estimate_dummy_outcome(func_args, action, new_outcome_train, X_train)
-                    new_outcome_train = estimator(X_train)
-                    new_outcome_validation = estimator(X_validation)
-                elif action == 'noise':
-                    new_outcome_train = self._noise(new_outcome_train, func_args)
-                    new_outcome_validation = self._noise(new_outcome_validation, func_args)
-                elif action == 'permute':
-                    new_outcome_train = self._permute(new_outcome_train, func_args)
-                    new_outcome_validation = self._permute(new_outcome_validation, func_args)
-                elif action =='zero':
-                    new_outcome_train = np.zeros(new_outcome_train.shape)
-                    new_outcome_validation = np.zeros(new_outcome_train.shape)
-            
-            new_data = validation_df.assign(dummy_outcome=new_outcome_validation)
+        simulation_results = []
+        refute_list = []
+
+        for _ in self._num_simulations:
+            no_estimator = self.check_for_estimator()
+            if no_estimator:
+                X_train = np.zeros(self._data[self._chosen_variables].shape)
+                outcome_train = np.zeros(self._data['y'].shape)
+                X_validation = self._data[self._chosen_variables].values
+                outcome_validation = self._data['y'].values
+
+                outcome_validation = self.process_data(X_train, outcome_train, X_validation, outcome_validation, self._transformations)

```

Comment:
do you need to define X_train here? I'm thinking that its enough to define X_validation and pass X_train as None.

Reply:
None works as well.